test_runner/random_test_runner/include/random_test_runner/randomizers.hpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2015-2020 Tier IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Developed by Robotec.ai.
+
+#ifndef RANDOM_TEST_RUNNER__RANDOMIZERS_H
+#define RANDOM_TEST_RUNNER__RANDOMIZERS_H
+
+#include <memory>
+#include <random>
+#include <type_traits>
+
+using RandomizationEngine = std::mt19937;
+using RandomizationEnginePtr = std::shared_ptr<RandomizationEngine>;
+
+template <typename T>
+class UniformRandomizer
+{
+public:
+  UniformRandomizer(RandomizationEnginePtr engine, T range_min, T range_max)
+  {
+    randomization_engine = std::move(engine);
+    randomization_distribution = std::make_unique<RandomizerDistributionType>(range_min, range_max);
+  }
+
+  void setRange(T range_min, T range_max)
+  {
+    randomization_distribution = std::make_unique<RandomizerDistributionType>(range_min, range_max);
+  }
+
+  T generate()
+  {
+    if (!randomization_engine) {
+      throw std::runtime_error("Randomizer not initialized");
+    }
+    return randomization_distribution->operator()(*randomization_engine);
+  }
+
+private:
+  RandomizationEnginePtr randomization_engine;
```

Comment:
It would be great to stick to the convention of suffixing `_`: `randomization_engine_` and `randomization_distribution_`